src/core/stores/VRange.ts

```diff
@@ -167,14 +170,16 @@ export class VRange {
      * @param reference
      */
     setStart(reference: VNode, position = RelativePosition.BEFORE): VRange {
-        reference = reference.firstLeaf();
-        if (position === RelativePosition.AFTER) {
-            reference.after(this.start);
-        } else {
-            reference.before(this.start);
-        }
-        this._direction = null; // Invalidate range direction cache.
-        return this;
+        return VDocument.withRange(() => {
+            reference = reference.firstLeaf();
+            if (position === RelativePosition.AFTER && reference !== this.end) {
```

Comment:
We can't set the start range as next sibling of the end range: that would be a backward collapsed range, which is not a thing in our abstraction (collapsed is always forward). I guess this is a fix but I think the problem only became visible with this commit.

Reply:
But this wouldn't happen if we don't do `withRange`, would it ? We really need to talk about the uses of `withRange`. I thought we would only use it when we specifically wanted to encounter the range, but here we are using it (for a reason I do not understand but it doesn't necessarily mean it's wrong) then, if we encounter it, we have a special if to protect us, which is specifically what I wanted to avoid in the first place.